doc/lsst.pipe.base/command-line-task-retargeting-howto.rst

```diff
@@ -0,0 +1,132 @@
+
+An example of the printed output is:
+
+.. code-block:: text
+
+   sub-tasks:
+   calibrate: lsst.pipe.tasks.calibrate.CalibrateTask
+   calibrate.applyApCorr: lsst.meas.base.applyApCorr.ApplyApCorrTask
+   calibrate.astromRefObjLoader: lsst.meas.algorithms.loadIndexedReferenceObjects.LoadIndexedReferenceObjectsTask
+   calibrate.astrometry: lsst.meas.astrom.astrometry.AstrometryTask
+   calibrate.astrometry.matcher: lsst.meas.astrom.matchOptimisticB.matchOptimisticBContinued.MatchOptimisticBTask
+   ...
+
+This subtask hierarchy is interpreted as follows:
+
+- The ``calibrate`` subtask of the command-line task (:command:`processCcd.py`) is configured to use `lsst.pipe.tasks.calibrate.CalibrateTask`.
+- ``calibrate`` (again, implemented by `~lsst.pipe.tasks.calibrate.CalibrateTask`) has a subtask named ``astrometry``, which is currently configured to use the `lsst.meas.astrom.astrometry.AstrometryTask` task.
+- ``calibrate.astrometry`` has a subtask named ``matcher``, which is implemented by `~lsst.meas.astrom.matchOptimisticB.matchOptimisticBContinued.MatchOptimisticBTask`.
+
+Note that if the ``calibrate.astrometry`` task is retargeted to a different task class, the subtask of ``calibrate.astrometry`` *may* change (for example, ``calibrate.astrometry.matcher`` may no longer exist).
+
+.. _command-line-task-retargeting-howto-configurablefield:
+
+How to retarget a subtask configured as a ConfigurableField with a configuration file
+=====================================================================================
+
+To retarget a subtask specified as an `lsst.pex.config.ConfigurableField`, you must use a configuration file (specified by :option:`--configfile`, see :ref:`command-line-task-config-howto-configfile`).
+Inside a configuration file, retargeting is done in two steps:
+
+1. Import the Task class you intend to use.
+2. Assign that class to to the subtask configuration using the `~lsst.pex.config.ConfigurableField.retarget` method.
+
+For example, to retarget the subtask named ``configurableSubtask`` with a class ``FooTask``, this configuration file should contain:
+
+.. code-block:: python
+
+   from ... import FooTask
+   config.configurableSubtask.retarget(FooTask)
+
+.. TODO make this a realistic example.
+
+Once a subtask is retargeted, you can configure it as normal by setting attributes to configuration parameters.
+For example:
+
+.. code-block:: python
+
+   config.configurableSubtask.subtaskParam1 = newValue
+
+.. warning::
+
+   When you retarget a task specified by an `lsst.pex.config.ConfigurableField` you lose all :ref:`configuration overrides <command-line-task-config-howto-obs>` for both the old and new task.
+   This limitation is not shared by `lsst.pex.config.RegistryField`.
+
+.. _command-line-task-retargeting-howto-registry-configfile:
+
+How to retarget a subtask configured as a RegistryField with a configuration file
+=================================================================================
+
```

Comment:
Please explain what a registry is, and how they are unrelated to data registries!